sycl-fusion/jit-compiler/include/Options.h

```diff
@@ -11,67 +11,23 @@
 
 #include "Kernel.h"
 
-#include <memory>
-#include <unordered_map>
-
 namespace jit_compiler {
 
 enum OptionID { VerboseOutput, EnableCaching, TargetDeviceInfo };
 
-class OptionPtrBase {};
-
-class Config {
-
-public:
-  template <typename Opt> void set(typename Opt::ValueType Value) {
-    Opt::set(*this, Value);
-  }
-
-  template <typename Opt> typename Opt::ValueType get() {
-    return Opt::get(*this);
-  }
-
-private:
-  std::unordered_map<OptionID, std::unique_ptr<OptionPtrBase>> OptionValues;
+struct OptionPtrBase {
+  OptionPtrBase(OptionID Id) : Id(Id) {}
 
-  void set(OptionID ID, std::unique_ptr<OptionPtrBase> Value) {
-    OptionValues[ID] = std::move(Value);
-  }
-
-  OptionPtrBase *get(OptionID ID) {
-    if (OptionValues.count(ID)) {
-      return OptionValues.at(ID).get();
-    }
-    return nullptr;
-  }
-
-  template <OptionID ID, typename T> friend class OptionBase;
+  const OptionID Id;
 };
 
-template <OptionID ID, typename T> class OptionBase : public OptionPtrBase {
-public:
+template <OptionID ID, typename T> struct OptionBase : public OptionPtrBase {
+  static constexpr OptionID Id = ID;
   using ValueType = T;
 
-protected:
-  static void set(Config &Cfg, T Value) {
-    Cfg.set(ID,
-            std::unique_ptr<OptionBase<ID, T>>{new OptionBase<ID, T>{Value}});
-  }
-
-  static const T get(Config &Cfg) {
-    auto *ConfigValue = Cfg.get(ID);
-    if (!ConfigValue) {
-      return T{};
-    }
-    return static_cast<OptionBase<ID, T> *>(ConfigValue)->Value;
-  }
+  OptionBase(T Val) : OptionPtrBase{ID}, Value{Val} {}
```

Comment:
```suggestion
  template <typename... Args>
  explicit OptionBase(Args&&... As) 
    : OptionPtrBase{ID}, Value{std::forward<Args>(As)...} {}
```
Building in-place might be a good idea for potential future users?

Reply:
Feel free to keep as is (add `explicit` maybe?)